dlt/common/destination/reference.py

```diff
@@ -201,21 +203,37 @@ def normalize_dataset_name(self, schema: Schema) -> str:
 
         If default schema name is None or equals schema.name, the schema suffix is skipped.
         """
-        if not schema.name:
+        dataset_name = self._make_dataset_name(schema.name)
+        return (
+            dataset_name
+            if not dataset_name
+            else schema.naming.normalize_table_identifier(dataset_name)
+        )
+
+    def normalize_staging_dataset_name(self, schema: Schema) -> str:
+        """Builds staging dataset name out of dataset_name and staging_dataset_name_layout."""
+        if "%s" in self.staging_dataset_name_layout:
+            # if dataset name is empty, staging dataset name is also empty
+            dataset_name = self._make_dataset_name(schema.name)
+            if not dataset_name:
```

Comment:
does this ever happen? I'm reading the code and trying to understand the mechanism, but in which case would the result of _make_dataset_name be None? Should't self.dataset_name alway be set?

Reply:
yes - in vector databases you can work on collections without dataset prefix. they override configuration making that field optional. btw. we should support that as well in clickhouse or dremio where we do not have schemas as well